lib/active_reporting/report.rb

```diff
@@ -49,24 +49,71 @@ def partition_dimension_filters(user_dimension_filter)
       end
     end
 
+    # Builds the sql statement to execute
+    #
+    # @return [String]
     def statement
-      parts = {
-        select: select_statement,
-        joins: dimension_joins,
-        group: group_by_statement,
-        having: having_statement,
-        order: order_by_statement
-      }
-
-      statement = ([model] + parts.keys).inject do |chain, method|
-        chain.public_send(method, parts[method])
-      end
+      case @metric.aggregate
+      when :sum
+        parts = {
+          select: select_statement,
+          joins: dimension_joins,
+          having: having_statement,
+          order: order_by_statement
+        }
+
+        statement = ([model] + parts.keys).inject do |chain, method|
+          chain.public_send(method, parts[method])
+        end
+
+        statement = process_scope_dimension_filter(statement)
+        statement = process_lambda_dimension_filter(statement)
+        statement = process_ransack_dimension_filter(statement)
+
+        # The original gem did not handle has_many relatinoships. In order to support
```

Comment:
Everything before the comment looks common to both cases, you could DRY this up a bit by moving those upwards and starting your case where the comment starts

Reply:
Oh good idea, will do that!